pkg/zio/zjsonio/reader.go

```diff
@@ -193,7 +193,7 @@ func decodeContainer(builder *zval.Builder, typ zeek.Type, body []interface{}) e
 		}
 		if columns != nil {
 			if k >= len(columns) {
-				return zng.ErrTypeMismatch
+				return &zng.RecordTypeError{Name: "<record>", Type: typ.String(), Err: zng.ErrExtraField}
```

Comment:
It seems like a look syntactic sugar here would be nice... maybe a helper function to create these error objects?

Reply:
`newRecordTypeError("record", typ.String(), zng.ErrExtraField)` is a few less characters but also adds one hop of indirection for the reader to see what is being returned.